Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/ExposureKeyHistoryCalculations.kt

```diff
@@ -1,39 +1,72 @@
 package de.rki.coronawarnapp.submission
 
+import KeyExportFormat
+import androidx.annotation.VisibleForTesting
 import com.google.android.gms.nearby.exposurenotification.TemporaryExposureKey
+import org.joda.time.Days
+import org.joda.time.Instant
 
 class ExposureKeyHistoryCalculations(
     private val transmissionRiskVectorDeterminator: TransmissionRiskVectorDeterminator,
+    private val daysSinceOnsetOfSymptomsVectorDeterminator: DaysSinceOnsetOfSymptomsVectorDeterminator,
     private val keyConverter: KeyConverter
 ) {
 
+    companion object {
+        const val MAX_DAYS_IN_PAST = 14
+        const val VECTOR_LENGTH = 15
+        const val TEN_MINUTES_IN_MILLIS = (10 * 60 * 1000).toLong()
+    }
+
     fun transformToKeyHistoryInExternalFormat(
         keys: List<TemporaryExposureKey>,
         symptoms: Symptoms
     ) =
         toExternalFormat(
-            toSortedHistory(limitKeyCount(keys)),
-            transmissionRiskVectorDeterminator.determine(symptoms)
+            limitKeyCount(toSortedHistory(keys)),
+            transmissionRiskVectorDeterminator.determine(symptoms),
+            daysSinceOnsetOfSymptomsVectorDeterminator.determine(symptoms, VECTOR_LENGTH)
         )
 
-    fun <T> limitKeyCount(keys: List<T>): List<T> =
-        keys.take(MAXIMUM_KEYS)
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun limitKeyCount(keys: List<TemporaryExposureKey>, now: Instant = Instant()) =
+        keys.filter { it.daysAgo(now) <= MAX_DAYS_IN_PAST }
```

Comment:
Ask: Please change the filter to `0 <= x <= MAX`, as `daysAgo`/`ageInDays` should never be negative but if it is for some reason (e.g. user manipulated the time on the device), we can eliminate potential errors. That's also how it's implemented on iOS.